SW-640 submodules moved, merged octoprint changes for prereleases

JFSolution opened the pull request.

Commit:
SW-640 submodules moved, merged octoprint changes for prereleases

JFSolution commented:
I didn't bump the version of the as it is not released yet

Yelko-mrbeam commented on docs/sw-update-conf.json:
```diff
@@ -15,28 +15,49 @@
+      "prerelease_channel": "mrbeam2-{tier}",
       "prerelease": true
     },
-    "beamos_date": {
-      "2021-06-11": {
-        "prerelease": true
-      }
```
What was this beamos_date used for and how? Why don't we need it anymore?

JFSolution replied:
this was used by the buster image, this used at the beginning a separate branch

Yelko-mrbeam reviewed (commented):
As far as I understand only the beamos_date was removed from everywhere and the modules were grouped under submodules where ever they are needed right? 
How come we don't need any change on the code? 🤔 
I guess to fully understand the changes I would need to know the code better 😄

Commit:
SW-640 moved the modules under the module key

amrsoll commented on docs/sw-update-conf.json:
```diff
@@ -14,110 +14,81 @@
+      "repo": "MrBeamPlugin",
+      "restart": "octoprint",
+      "pip": "https://github.com/mrbeam/MrBeamPlugin/archive/{target_version}.zip",
+      "submodules": {
```
Do you mean "dependencies" ?

amrsoll replied:
And what is the goal with grouping them under the mrb plugin? they are inherently indepedent and installed in a different venv

JFSolution replied:
hmm, yeah this can be a better name

JFSolution replied:
these will be installed via the update_script and not by octoprint itselfe

amrsoll replied:
so only this "submodule" category gets passed to an installer script?

JFSolution replied:
that is the plan

Yelko-mrbeam replied:
are we changing the name to dependencies or leaving submodules @Josef-MrBeam? 🤔

Yelko-mrbeam replied:
ah sorry I see it was already changed 🤦

Commit:
SW-640 changed "submodules" to "dependencies"

khaledsherkawi commented on docs/sw-update-conf.json:
```diff
@@ -14,110 +14,81 @@
+          "pip": "https://github.com/mrbeam/netconnectd_mrbeam/archive/{target_version}.zip",
+          "global_pip_command": true,
+          "package_name": "netconnectd",
+          "branch": "mrbeam2-stable"
```
Why is the branch mentioned here but not in the other dependencies?

JFSolution replied:
because this branch will always be mrbeam2-stable for all tiers of the legacy image, but I see now that I didn't handle the buster image case, I have to add this

khaledsherkawi commented on docs/sw-update-conf.json:
```diff
@@ -14,110 +14,81 @@
-        "branch": "mrbeam2-{tier}-buster",
-        "develop": {
-          "branch": "develop"
+    },
```
Can we assume here that stable will pull from the release directly but dev, alpha, and beta according to the above configuration?

JFSolution replied:
if there is a additional configuration for a tier this will override the dedault config of the module

khaledsherkawi commented on docs/sw-update-conf.json:
```diff
@@ -14,110 +14,81 @@
+          "repo": "mrb_hw_info",
+          "pip": "git+ssh://git@bitbucket.org/mrbeam/mrb_hw_info.git@{target_version}",
+          "global_pip_command": true,
+          "package_name": "mrb-hw-info",
```
Why is the "package_name" key added here and in Netconnectd Daemon but not for iobeam and mrbeam-ledstrips?

JFSolution replied:
because it is different to the repo name

khaledsherkawi reviewed (changes requested):
Buster image configuration to be added

Commit:
SW-640 added branch defaults and venv pip commands

Commit:
move MrBeamDoc in MrBeam Plugin submodules and remove versionnumber inside the file

Yelko-mrbeam commented on docs/sw-update-conf.json:
```diff
@@ -80,7 +102,13 @@
+          "beamos_date": {
+            "2021-06-11": {
+              "pip_command": "sudo /usr/local/netconnectd/venv/bin/pip",
+              "branch": "master"
```
Do we need to add the branch somewhere else? is there a default value?

JFSolution replied:
what do you mean? This will override the mrbeam2-stable that is used for the legacy image. But you are right I should check if this be handeld correct for the alpha, beta, develop branch for the buster image

Yelko-mrbeam commented on docs/sw-update-conf.json:
```diff
@@ -1,5 +1,4 @@
 {
-  "version": "1",
```
why is this version removed? who is using this?

JFSolution replied:
this was used before, now I changed it to use the release/tag version instead

Yelko-mrbeam commented on docs/sw-update-conf.json:
```diff
@@ -1,123 +1,121 @@
+          "pip": "git+ssh://git@bitbucket.org/mrbeam/iobeam.git@{target_version}",
+          "global_pip_command": true,
+          "api_user": "MrBeamDev",
+          "api_password": "v2T5pFkmdgDqbFBJAqrt",
```
Is it fine to have the password here? shouldn't it be somewhere more safe?

JFSolution replied:
but where? before it was also open in the MrBeamPlugin repo

Yelko-mrbeam replied:
I guess it would be anyway a different ticket.. We could bring it up later on...

Commit:
fix netconnectd submodule config

Commit:
change Mr Beam Plugin to update with update script

Commit:
change mrbeam plugin to release based updates

Commit:
update netconnect plugin config to use update_script
add update_script_relative_path for update script methods
remove unused values of dependencies

Commit:
fix netconnectd config

Commit:
only use update script on alpha and develop, beta and stable stay the old way

Commit:
fix develop for mrbeam plugin

Commit:
develop tier commit based from alpha branch
alpha tier release based
beta tier commit based from mrbeam2-beta
stable tier commit based from mrbeam2-stable

Commit:
change restart of dependency dontaining modules
remove branches of legacy image for netconnectd

Yelko-mrbeam commented on docs/sw-update-conf.json:
```diff
@@ -1,123 +1,156 @@
+        },
+        "mrbeamdoc": {
+          "repo": "MrBeamDoc",
+          "pip": "https://github.com/mrbeam/MrBeamDoc/archive/{target_version}.zip"
```
no need to add beamos_date?

JFSolution replied:
The beamos date is only needed if you need a seperate config for the given date
